### PR TITLE
Add tip 70: Flutter gen localization

### PR DESCRIPTION
**Pull request by pr-Mais**
This tip shows how to localize a Flutter app with the Flutter tool, without using third-party packages.

### CODE REVIEW

File: tips/0070-localizations/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+Add a new key to `l10n.yaml`: `synthetic-package: false`. This will let Flutter knows we need a non-synthetic package, it is true by default.
+
+![](070.6.png)
+
+---
+
+Let’s explore the generated files 🔍
+
+Open the main file, `app_localizations.dart`, we can notice the key `name` from the ARB template, along with the description. The `AppLocalizations` class is an abstract class that will be extended by every locale delegate.
```
Comment (bizz84) [inline tips/0070-localizations/index.md:65]: *Break in two lines:*

Inside the main file (`app_localizations.dart`), we can notice the key `name` from the ARB template, along with the description.

The `AppLocalizations` class is an abstract class that will be extended by every locale delegate.

File: tips/0070-localizations/index.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+👉🏻 `app_localizations_en.dart`
+👉🏻 `app_localizations_es.dart`
+
+As noted earlier, both extend the base class `AppLocalizations`.
+
+![](070.8.png)
+
+---
+
+5️⃣ Finally, we need to add the localizations delegate to our `MaterialApp`. We can then use it anywhere in the app!
```
Comment (bizz84) [inline tips/0070-localizations/index.md:82]: *Break in two lines, add emoji*

Finally, we need to add the localizations delegate to our `MaterialApp`.

We can then use it anywhere in the app! 🚀

File: tips/0070-localizations/index.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+![](070.9.png)
+
+---
+
+Extra: BuildContext extension 🧩
```
Comment (bizz84) [inline tips/0070-localizations/index.md:88]: Replace `Extra` with `Bonus`

File: tips/0070-localizations/index.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+This is not a required step, yet it’s a convenient way to make the syntax shorter.
```
Comment (bizz84) [inline tips/0070-localizations/index.md:90]: *Rewrite as:*

This is not required, but helps us make the syntax shorter.

File: tips/0070-localizations/index.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+![](070.10.png)
+
+---
+
+You can find more in the official Flutter documentation 👇🏻
```
Comment (bizz84) [inline tips/0070-localizations/index.md:96]: _Let's send people back to my article instead_

To learn more about localization, read this:

https://codewithandrea.com/articles/flutter-localization-build-context-extension/

File: tips/0070-localizations/index.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
+# Localizing Flutter apps with code gen
+
+Did you know that you can set up your Flutter app localizations in 5 minutes, using Flutter's tool generation? 🗺
+
+Let’s start by adding the required packages: `flutter_localizations` and `intl`.
```
Comment (bizz84) [inline tips/0070-localizations/index.md:5]: Minor nitpick: strictly speaking, I think `intl` is not necessary.

Any specific reason you included it here?

**Review by bizz84 — CHANGES_REQUESTED** [review]

File: tips/0070-localizations/index.md (hunk 6 above)
Comment (pr-Mais) [reply tips/0070-localizations/index.md:5]: It's not, but then the analyzer will leave a warning.

Screen Shot 2022-09-25 at 12 50 44 PM

Comment (bizz84) [reply tips/0070-localizations/index.md:5]: Ok got it

**Comment by bizz84** [comment]
@pr-Mais the thread content is good, but it's missing a strong hook telling people why they should care about this.

I've added `070.0` to Figma. Proposed intro tweet:

```
If your users speak another language, you’ll need to localize your Flutter app.

Here’s how to setup Flutter app localizations in just 5 minutes (using code generation).

🧵
```

File: tips/0070-localizations/index.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
+# Localizing Flutter apps with code gen
```
Comment (bizz84) [inline tips/0070-localizations/index.md:1]: Let's start with a different tweet:

```
If your users speak another language, you’ll need to localize your Flutter app.

Here’s how to setup Flutter app localizations in just 5 minutes (using code generation).

🧵
```

Then use `070.0` as the image.

File: tips/0070-localizations/index.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+Did you know that you can set up your Flutter app localizations in 5 minutes, using the Flutter generation tool? 🗺
```
Comment (bizz84) [inline tips/0070-localizations/index.md:3]: Given that we have a new intro tweet, we could rewrite this as:

```
1️⃣  As a first step, we need to add the required packages to the pubspec.yaml file.
```

File: tips/0070-localizations/index.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+Let’s start by adding the required packages: `flutter_localizations` and `intl`.
+
+🧵
+
+![](070.1.png)
+
+---
+
+1️⃣ Create `l10n.yaml` at the root
```
Comment (bizz84) [inline tips/0070-localizations/index.md:13]: Change to 2️⃣

File: tips/0070-localizations/index.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+✔️ `arb-dir` is the directory where Flutter will look for the localized strings
+✔️ Dart localizations classes will be generated in the `output-dir`
+✔️ `template-arb-file` is the main template that contains a description for each localized message
+
+![](070.2.png)
+
+---
+
+2️⃣ Create the ARB files under lib
```
Comment (bizz84) [inline tips/0070-localizations/index.md:23]: Change to 3️⃣

File: tips/0070-localizations/index.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+An ARB file is an “application resource bundle” used by Flutter to generate corresponding Dart files. Each locale will have its own ARB file.
+
+For example, to add French, just create app_fr.arb in the same directory.
+
+![](070.3.png)
+
+---
+
+3️⃣ Add the `@@locale` key and the first key/value pair. Note that there is a small difference between the file we declared as a template in l10n.yaml and other ARB files.
```
Comment (bizz84) [inline tips/0070-localizations/index.md:33]: Change to 4️⃣

File: tips/0070-localizations/index.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+Synthetic or non-synthetic?
+
+The localizations files can either be generated as a synthetic or non-synthetic package.
+
+Let's see what this means and which one we should choose.👇🏻
+
+---
+
+4️⃣: Synthetic
```
Comment (bizz84) [inline tips/0070-localizations/index.md:47]: Change to 5️⃣

File: tips/0070-localizations/index.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+Set `generate` to true in pubspec.yaml, then run the command `flutter gen-10n` at the root using the terminal. This will generate a new package under .dart_tool, which ignores the `output-dir` we specified earlier in `l10n.yaml`.
+
+![](070.5.png)
+
+---
+
+4️⃣: Non-synthetic
```
Comment (bizz84) [inline tips/0070-localizations/index.md:55]: Change to 5️⃣

File: tips/0070-localizations/index.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+👉🏻 `app_localizations_en.dart`
+👉🏻 `app_localizations_es.dart`
+
+As noted earlier, both extend the base class `AppLocalizations`.
+
+![](070.8.png)
+
+---
+
+5️⃣ Finally, we need to add the localizations delegate to our `MaterialApp`.
```
Comment (bizz84) [inline tips/0070-localizations/index.md:88]: Change to 6️⃣

**Comment by pr-Mais** [comment]
Done

**Pull request merged**